crates/bevy_ecs/examples/derive_label.rs

```diff
@@ -0,0 +1,39 @@
+use bevy_ecs::prelude::*;
+
+fn main() {
+    // Unit labels are always equal.
+    assert_eq!(UnitLabel.as_label(), UnitLabel.as_label());
```

Comment:
These tests are critical, but won't be run in CI. I actually think that these should be moved to doc tests on `SystemLabel`.

Reply:
I agree, but we can't actually add doctests to label types until #5367